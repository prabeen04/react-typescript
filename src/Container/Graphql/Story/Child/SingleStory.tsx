import * as React from 'react';
import * as types from '../StoryActionTypes';
import { IStory } from '../StoryInterface';

interface ISingleStory {
    story: IStory;
}
export default function SingleStory(props: ISingleStory): JSX.Element {
    return (
        <h1
            data-test="SingleStory"
            className='story-title'
            onClick={dispatch({ type: types.SET_SELECTED_STORY, payload: 'test' })}
        >
            {props.story.title}
        </h1>)
}