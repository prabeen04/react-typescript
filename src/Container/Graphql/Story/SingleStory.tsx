import * as React from 'react';
import { IStory } from './StoryInterface';

export default function Story(props: IStory): JSX.Element {
    return <h1 className='story-title' data-test="Story">{props.title}</h1>
}