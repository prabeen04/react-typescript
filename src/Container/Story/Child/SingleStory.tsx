import * as React from 'react';
import { withRouter } from "react-router-dom";
import { RouteComponentProps } from "react-router";
import moment from 'moment';
import { useMutation } from "react-apollo-hooks";
import * as types from '../StoryActionTypes';
import { IStory } from '../StoryInterface';
import { DELETE_STORY, GET_STORIES } from "../StoryQuery";
import StoryContext from '../StoryContext'

interface ISingleStory extends RouteComponentProps<any> {
    story: IStory;

}
const SingleStory = (props: ISingleStory): JSX.Element => {
    const { state, dispatch } = React.useContext(StoryContext)
    function handleStoryDetailpush() {
        dispatch({ type: types.SET_SELECTED_STORY, payload: props.story.id })
        props.history.push({
            pathname: `/story/${props.story.id}`,
            state: { story: props.story }
        })
    }
    function handleDeleteStory(e: React.SyntheticEvent, deleteStory: any, id: string) {
        e.stopPropagation();
        let allowDelete: any = confirm("Are you sure ?")
        if (allowDelete) {
            console.log('after confirm')
            deleteStory({
                variables: {
                    storyId: id
                }
            })
        }
    }
    function handleUpdateStory(e: React.SyntheticEvent, id: string) {
        e.stopPropagation();
        dispatch({
            type: types.SET_EDITING_STATE,
            payload: {
                isEditing: true,
                addStoryModal: true,
                currentEditingStory: props.story,
            }
        })
    }
    function updateStoriesAfterDelete(cache: any, { data: { deleteStory } }: any) {
        const { stories } = cache.readQuery({ query: GET_STORIES });
        console.log(stories, deleteStory)
        const updatedStories = stories.filter((story: IStory) => story.id !== deleteStory.id)
        cache.writeQuery({
            query: GET_STORIES,
            data: { stories: updatedStories },
        });
    }
    const deleteStory = useMutation(DELETE_STORY, { variables: { storyId: props.story.id }, update: updateStoriesAfterDelete })

    return (
        <>
            <div className="single-story"
                onClick={handleStoryDetailpush}
            >
                <i className="fas fa-book-reader"></i>
                <div style={{ flexBasis: '95%' }}>
                    <h1
                        data-test="SingleStory"
                        className='story-title'
                    >
                        {props.story.title}
                    </h1>
                    <p className='story-time'>{moment(props.story.createdAt).format('ll')}</p>
                </div>
                <div className="story-action-wrapper">
                    <i className="fas fa-edit"
                        onClick={(e: React.SyntheticEvent) => handleUpdateStory(e, props.story.id)}
                    ></i>
                    <i className="fas fa-trash-alt"
                        onClick={(e: React.SyntheticEvent) => handleDeleteStory(e, deleteStory, props.story.id)}
                    ></i>
                </div>
            </div>
        </>
    )
}
export default withRouter(SingleStory)