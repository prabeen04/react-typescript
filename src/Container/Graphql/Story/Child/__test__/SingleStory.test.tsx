import * as React from 'react';
import { shallow } from "enzyme";
import SingleStory from '../SingleStory';

describe('<SingleStory/>', () => {
    const props = {
        title: 'test'
    }
    it('should match snapshot', () => {
        const wrapper = shallow(<SingleStory {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
    it('should render story', () => {
        const wrapper = shallow(<SingleStory {...props} />)

        expect(wrapper.find('[data-test="SingleStory"]').length).toEqual(1)
    })
})
})
