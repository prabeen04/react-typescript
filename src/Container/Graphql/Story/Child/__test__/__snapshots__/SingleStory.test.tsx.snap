// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleStory/> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Story
    title="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "test",
      "className": "story-title",
      "data-test": "Story",
    },
    "ref": null,
    "rendered": "test",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
        "className": "story-title",
        "data-test": "Story",
      },
      "ref": null,
      "rendered": "test",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
